### ISSUE TITLE
CalendarView : The present day in the calendar is represented with different color but the same is not informed to the screen reader users.

### ISSUE THREAD

**Issue by MayuriNaik** [microsoft/microsoft-ui-xaml#1590]
GitHubTags: #A11y_OSvisualupdate;#A11yMAS;#A11yTCS;#Win32;#DesktopApp;#WindowsUILibrary;#MAS1.3.1;#MAS4.2.1;#Narrator;#NVDA; #A11ySev2;#Benchmark

**Environment Details:**
Application Name: OS visual update
Windows Version: Windows10
Screen Reader: Narrator, NVDA 2019.2

**Steps to Reproduce**:
1. Launch Mini Control Gallery App from the installer location.
2. Screen with all generic controls would appear on left side of the screen.
3. Navigate to "Generic Controls" section.
4. Start screen reader.
5. Navigate to the present date and listen if the same information is announced to screen reader.

**Actual:**
When screen reader users navigate to the calendar the present date is represented by change of color visually, but this information is not announced to the screen reader users.

**Similar issue is observed in the below scenario:**
1. MiniControlGallery - Calender and Color Picker Controls -> Launch Mini Control Gallery App from the installer location. -> Screen with Calender and Color Picker controls would appear on right side of the screen.

**Expected:**
The information that is available for the visual users should also be announced to the screen reader users.

**User Impact:**
If some content is only represented by the change of color visually then the screen reader users will not get the information that is available for visual users.

**Attachment:**
![A11y_ControlUIupdatebasedonSantoriniOSvisualupdate_MiniControGallery_GenericControls_Screenreader  The present day in the calendar is represented with different color but the same is](https://user-images.githubusercontent.com/52036772/68754559-f0699380-062c-11ea-8fe0-f20737c5c565.gif)

**Proposed Fix**
[Edited by @YuliKl]
Proposed in an [internal discussion](https://stackoverflow.microsoft.com/questions/174560/whats-the-correct-uia-pattern-to-express-today-in-the-calendarview-control/176026), the suggestion is to update the UIA name of the today cell to include the localized word "today" in addition to the date.

**Comment by jevansaks** [comment]
@kikisaints @yulikl This has been an issue for a while, we can fix this but we need to know where we should actually encode this information in the UIA properties. Should the element with the current day be amended to have this in the "Name"? That seems weird but maybe...

**Comment by kikisaints** [comment]
@jevansaks I believe we do something similar with checked radiobuttons/checkboxes. The screen reader says "<label> CheckBox, selected" - could we not follow the same pattern here?

**Comment by jevansaks** [comment]
But it's not "selected" (there's another date that is "selected"). It's the "Today" date that has special visual treatment but appears the same via UIA. Is there another automation property we could set, perhaps?

**Comment by kikisaints** [comment]
I'm not sure, seems like a question for @YuliKl... but going out on a limb here, there appears to be an [itemstatusproperty](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.automationproperties.itemstatusproperty#Windows_UI_Xaml_Automation_AutomationProperties_ItemStatusProperty) available in the automation properties class. Could that be used?

**Comment by YuliKl** [comment]
Added a **proposed fix** into the issue description:
> Proposed in an [internal discussion](https://stackoverflow.microsoft.com/questions/174560/whats-the-correct-uia-pattern-to-express-today-in-the-calendarview-control/176026), the suggestion is to update the UIA name of the today cell to include the localized word "today" in addition to the date.

**Comment by mdtauk** [comment]
> Added a **proposed fix** into the issue description:
> 
> > Proposed in an [internal discussion](https://stackoverflow.microsoft.com/questions/174560/whats-the-correct-uia-pattern-to-express-today-in-the-calendarview-control/176026), the suggestion is to update the UIA name of the today cell to include the localized word "today" in addition to the date.

"Today" or "Today's Date", basically consideration for the context in what the picker scenario is.  And would there also be the concept of "Tomorrow" and "Yesterday"?

**Comment by YuliKl** [comment]
> And would there also be the concept of "Tomorrow" and "Yesterday"?

I don't think so. The issue here is that sighted users get information (in the form of an accent-colored background) that's not available to blind users. The lack of equity needs to be corrected. But we don't color-code tomorrow or yesterday, so shouldn't provide this information to ATs.

**Comment by mdtauk** [comment]
> > And would there also be the concept of "Tomorrow" and "Yesterday"?
> 
> I don't think so. The issue here is that sighted users get information (in the form of an accent-colored background) that's not available to blind users. The lack of equity needs to be corrected. But we don't color-code tomorrow or yesterday, so shouldn't provide this information to ATs.

I was thinking more about context in how it is announced.  If you are picking a date, then announcing Today instead of Today's Date may feel odd?
